Lib/axisregistry/__init__.py

```diff
@@ -1,52 +1,458 @@
+        axes_in_font = {
+            a.axisTag: {"min": a.minValue, "max": a.maxValue}
+            for a in ttFont["fvar"].axes
+        }
+        for axis in self:
+            if axis not in axes_in_font:
+                log.warn(f"Axis {axis} not found in GF Axis Registry!")
+                continue
+            for fallback in self[axis].fallback:
+                if (
+                    fallback.value < axes_in_font[axis]["min"]
+                    or fallback.value > axes_in_font[axis]["max"]
+                ):
+                    continue
+                res[axis].append(fallback)
+        return res
+
+    def fallbacks_in_name_table(self, ttFont):
+        res = []
+        name_table = ttFont["name"]
+        tokens = (
+            name_table.getBestFamilyName().split()
+            + name_table.getBestSubFamilyName().split()
+        )
+        fvar_axes_in_font = [a.axisTag for a in ttFont["fvar"].axes]
+        for token in tokens:
+            axis, fallback = axis_registry.get_fallback(token)
+            if not axis or axis in fvar_axes_in_font:
+                continue
+            res.append((axis, fallback))
+        return res
+
+
+axis_registry = AxisRegistry()
+
+
+def is_variable(ttFont):
+    return "fvar" in ttFont
+
+
+def _fvar_dflts(ttFont):
+    res = OrderedDict()
+    for a in ttFont["fvar"].axes:
+        # find name and elision
+        if a.axisTag in axis_registry:
+            name = next(
+                (
+                    f.name
+                    for f in axis_registry[a.axisTag].fallback
+                    if f.value == a.defaultValue
+                ),
+                None,
+            )
+            elided = a.defaultValue == axis_registry[
+                a.axisTag
+            ].default_value and name not in ["Regular", "Italic"]
+        else:
+            name = None
+            elided = True  # since we can't find a name for it, keep it elided
+
+        res[a.axisTag] = {"value": a.defaultValue, "name": name, "elided": elided}
+    return res
+
+
+def build_stat(ttFont, sibling_ttFonts=None):
+    log.info("Building STAT table")
+    assert is_variable(ttFont), "not a VF!"
+    fallbacks_in_fvar = axis_registry.fallbacks_in_fvar(ttFont)
+    fallbacks_in_siblings = list(
+        chain.from_iterable(
+            axis_registry.fallbacks_in_name_table(f) for f in sibling_ttFonts
+        )
+    )
+    fallbacks_in_names = axis_registry.fallbacks_in_name_table(ttFont)
+    nametable = ttFont["name"]
+
+    # rm old STAT table and associated name table records
+    if "STAT" in ttFont:
+        stat = ttFont["STAT"]
+        axis_values = stat.table.AxisValueArray.AxisValue
+        axes = stat.table.DesignAxisRecord.Axis
+        for ax in axis_values:
+            nametable.removeNames(nameID=ax.ValueNameID)
+        for ax in axes:
+            nametable.removeNames(nameID=ax.AxisNameID)
+        del ttFont["STAT"]
+
+    res = []
+    # use fontTools build_stat
```

Comment:
buildStatTable. And I would expand that comment to say that the URL explains the expected format of the argument to buildStatTable.